qiskit/pulse/utils.py

```diff
@@ -65,6 +64,19 @@ def format_parameter_value(
             evaluated = float(evaluated.real)
             if evaluated.is_integer():
                 evaluated = int(evaluated)
+        else:
+            warnings.warn(
```

Comment:
@Eric-Arellano Can we use your deprecation decorator for this case? Seems like `predicate` is good place to do type check, but writing try-except logic inside lambda function looks bit ugly.

Reply:
Yeah this is what I'm thinking of as an alternative. But writing another function for just checking type seems too much overhead, and indeed inlining the warning is much simpler. I know your decorator properly updates API docs. Does it make sense to suppress exception here?
https://github.com/Qiskit/qiskit-terra/blob/6a946cf47886c3d6de4b52c98c3bacbda0d35df8/qiskit/utils/deprecation.py#L291-L292

Reply:
> But writing another function for just checking type seems too much overhead

What's the overhead? Code readability?

> Note that this function is not added to toctree and we don't have motivation for using decorator for the API doc update.

The decorator has two benefits:
1. Update the docstring for the function so that it shows up in API docs and `help(my_func)` in REPLs/Jupyter
2. Standardize the message

Reply:
> Again, this function is not exposed to end-users and documentation doesn't become good motivation for us, though standardization of error massage is worth considering

Here's the standard format:

https://github.com/Qiskit/qiskit-terra/blob/751fa46947aa1cc581b0fc25673236e396397b0f/test/python/utils/test_deprecation.py#L152-L155

You'd want to change the first part of the first sentence to say something like 

> Passing a complex to the argument ``arg5`` in the function ``blah`` is pending...

--

> Currently I tend to approve this PR as it is if you don't have strong objection.

No objections from me to approve the PR as is :) The main thing I care about is docs having the deprecations, which isn't relevant here.

> It would be great in future if the deprecate util could support such (edge) case more efficiently.

Hm, idea: https://github.com/Qiskit/qiskit-terra/issues/9902

Reply:
Thanks @Eric-Arellano for the detailed explanation!